### DISCUSSION TITLE
Query method not valid

### DISCUSSION THREAD

**Discussion by ghost** [appwrite/appwrite#5804] [Q&A] [answered]
### 💭 Description

- I have followed the documentation but the result is not correct
address: https://appwrite.io/docs/databases-queries

- My code

```javascript
const { Client, Databases, ID, Query } = require('appwrite');

const client = new Client().setEndpoint('https://cloud.appwrite.io/v1').setProject('64ad1e3103fde8825ab8');

const databases = new Databases(client);

const promise = databases.listDocuments('64ad1e91ce62e8e6297f', '64ad1eb651ec1c154276', [
    /* I want to get data with 'name' attribute and user read permission status is on */
    Query.select(['name'])
]);

promise.then(
  function (response) {
    console.log(response);
  },
  function (error) {
    console.log(error);
  }
);
```

- I get an error message

```javascript
AppwriteException: Query method not valid: select
    at Client.<anonymous> (C:\Users\myuser\Desktop\myapp\node_modules\appwrite\dist\cjs\sdk.js:393:27)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\myuser\Desktop\myapp\node_modules\appwrite\dist\cjs\sdk.js:24:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 400,
  type: 'general_argument_invalid',
  response: {
    message: 'Query method not valid: select',
    code: 400,
    type: 'general_argument_invalid',
    version: '0.10.45'
  }
}
```

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment, marked as answer]
@hthuantj, thanks for raising this! 🙏🏼 

The SDK you're using is meant for Appwrite 1.3.x, but you seem to be connecting to Appwrite Cloud, which is on 1.1.2. That's why Appwrite is rejecting your request with the error mentioned. For Appwrite Cloud, you should use version 10.1.0 of the appwrite NPM package.

**Reply by ghost** [reply]
Thank you.
